automation/build.sh

```diff
@@ -25,99 +24,97 @@
 # to the docker registry.
 # If CLEANUP is "true", all images will be removed after pushing - including any relevant images
 # that may have been on the host from before the build, so be careful!
-# If ENABLE_TESTS is "true", tests will be run.
 #
-# Requires docker >= 17.05, and make.
+# Requires docker >= 17.05
 #
 
 set -e
 
 THIS_FILE=$0
 if [ -z "$ARCH" ] || [ -z "$TAG" ]; then
-	cat $THIS_FILE | awk '{if(/^#/)print;else exit}' | tail -n +2 | sed 's/\#//'
+	awk '{if(/^#/)print;else exit}' "${THIS_FILE}" | tail -n +2 | sed 's/\#//'
 	exit 1
 fi
 
 # This is the supervisor image we will produce
-TARGET_IMAGE=balena/$ARCH-supervisor:$TAG$DEBUG
+TARGET_IMAGE=balena/$ARCH-supervisor:$TAG
+TARGET_BUILD_IMAGE=balena/$ARCH-supervisor:$TAG-build
 
-# Intermediate images and cache
-NODE_IMAGE=balena/$ARCH-supervisor-node:$TAG$DEBUG
-NODE_BUILD_IMAGE=balena/$ARCH-supervisor-node:$TAG-build$DEBUG
-
-TARGET_CACHE_MASTER=balena/$ARCH-supervisor:master$DEBUG
-NODE_CACHE_MASTER=balena/$ARCH-supervisor-node:master$DEBUG
-NODE_BUILD_CACHE_MASTER=balena/$ARCH-supervisor-node:master-build$DEBUG
+MASTER_IMAGE=balena/$ARCH-supervisor:master
+MASTER_BUILD_IMAGE=balena/$ARCH-supervisor:master-build
 
 CACHE_FROM=""
 function useCache() {
 	image=$1
 	# Always add the cache because we can't do it from
 	# a subshell and specifying a missing image is fine
-	CACHE_FROM="$CACHE_FROM --cache-from $image"
+	CACHE_FROM="--cache-from $image ${CACHE_FROM}"
 	# Pull in parallel for speed
-	docker pull $image &
+	docker pull "$image" &
 }
 
-useCache $TARGET_IMAGE
-useCache $TARGET_CACHE_MASTER
-useCache $NODE_IMAGE
-useCache $NODE_CACHE_MASTER
-# Debug images don't include nodebuild or use the supervisor-base image
-if [ -z "$DEBUG" ]; then
-	useCache $NODE_BUILD_IMAGE
-	useCache $NODE_BUILD_CACHE_MASTER
-
-	docker pull balenalib/amd64-node:6-build &
-	docker pull balena/$ARCH-supervisor-base:v1.4.7 &
-fi
-# Pull images we depend on in parallel to avoid needing
-# to do it in serial during the build
-docker pull balenalib/raspberry-pi-node:10-run &
-docker pull balenalib/armv7hf-node:10-run &
-docker pull balenalib/aarch64-node:10-run &
-docker pull balenalib/amd64-node:10-run &
-docker pull balenalib/i386-node:10-run &
-docker pull balenalib/i386-nlp-node:6-jessie &
-wait
+function retryImagePush() {
+	local image=$1
+	local retries=0
```

Comment:
```suggestion
	local -i retries
```

will ensure this is an int, and set to 0